### ISSUE TITLE
Siamese model not learning anything.

### ISSUE THREAD

**Issue by sainimohit23** [keras-team/keras#12901]
I have a large dataset of handwritten signatures. I constructed a siamese model for my dataset as described here: https://github.com/keras-team/keras/blob/master/examples/mnist_siamese.py .

Now, when I train my model. Loss does not improve after first epoch. Model is always giving 0 as output.

**Comment by mani2106** [comment]
What is the size of the image that you are passing to the network?
Can you try changing the model architecture with some convolutions?

**Comment by sainimohit23** [comment]
@mani2106 I have tried 2-3 architectures. None of them are working. I also tried the models like vgg and inception. Nothing is working. The number of pairs in the training dataset is around 2,50,000.

Image size= 155x220x3.

When I use binary_crossentropy or sigmoid on the difference of image encodings. Then model is working very well.

**Comment by mani2106** [comment]
Have you tried grayscaling the images?
I don't know VGG or Inception will be actually useful for signatures.
The example given was done with very small images `28 x 28` so that architecture may not be suitable for other images

**Comment by sainimohit23** [comment]
@mani2106 I have tried everything that you have mentioned. But, still, I am not getting results. I have also tried triplet loss. But, triplet loss and contrastive loss end up encoding images into a vector of zeros.

Sometimes, `binary_crossentropy` also ends up predicting zeros for every input. So, I have to retrain the model.

**Comment by mani2106** [comment]
Have you also tried changing the `margin` of the `contrastive loss` function, I have had some improvements in my work with that change
